enhancements/20220525-system-logging.md

```diff
@@ -0,0 +1,174 @@
+you can use the following yamls:
+
+```yaml
+# graylog-cluster-flow.yaml
+apiVersion: logging.banzaicloud.io/v1beta1
+kind: ClusterFlow
+metadata:
+  name: "all-logs-gelf-hs"
+  namespace: "cattle-logging-system"
+spec:
+  globalOutputRefs:
+    - "example-gelf-hs"
+```
+
+```yaml
+# graylog-cluster-output.yaml
+apiVersion: logging.banzaicloud.io/v1beta1
+kind: ClusterOutput
+metadata:
+  name: "example-gelf-hs"
+  namespace: "cattle-logging-system"
+spec:
+  gelf:
+    host: "192.168.122.159"
+    port: 12202
+    protocol: "udp"
+```
+
+You can verify that they are installed successfully, you can check them with `kubectl`:
+
+```bash
+>>> kubectl get clusteroutputs -n cattle-logging-systems example-gelf-hs
+NAME              ACTIVE  PROBLEMS
+example-gelf-hs   true
+>>> kubectl get clusterflows -n cattle-logging-systems example-gelf-hs
+NAME                ACTIVE  PROBLEMS
+all-logs-gelf-hs    true
+```
+
+#### Viewing Logs Via Grafana Loki
+
+### Setup
+
+Loki will not be installed to the cluster by default, but you can manually install it:
+
+  1. Install the helm chart: `helm install --repo https://grafana.github.io/helm-charts --values enhancements/20220525-system-logging/loki-values.yaml --version 2.7.0 loki-stack`s
+  2. Apply the `ClusterFlow` and `ClusterOutput`: `kubectl apply -f enhancements/20220525-system-logging/loki.yaml`
+
+After some time for the logging operator to load and apply the `ClusterFlow` and `ClusterOutput` you will see the logs
+flowing into loki.
+
+##### Accessing
+
+By default, harvester routes logs to its internal Grafana Loki deployment. To view the Loki UI, you can go to port
+forward to the `loki-stack-grafana` service. For example map `localhost:3000` to `loki-stack-grafana`s port 80:
+
+```bash
+kubectl port-forward --namespace cattle-logging-system service/loki-stack-grafana 3000:80
+```
+
+##### Authentication
+
+The default username is admin. To get the password for the admin user run the following command:
+
+```bash
+kubectl get secret --namespace cattle-logging-system loki-stack-grafana --output jsonpath="{.data.admin-password}" | base64 --decode
+```
+
+##### Log Querying
+
+Once the UI is open on the left tab you can select the "Explore" tab to take you to a page where you manually send
+queries to loki:
+
+![](./20220525-system-logging/loki-explore-tab.png)
+
+In the search bar you can enter queries to select the logs you are interested in. The query language is described in the
+[grafana loki docs](https://grafana.com/docs/loki/latest/logql/log_queries/). For example, to select all logs in the
+`cattle-logging-system`:
+
+![](./20220525-system-logging/loki-explore-query-kubernetes-namespace-name-cattle-logging-system.png)
+
+Or to select harvester host machine logs from the `rke2-server.service` service:
+
+![](./20220525-system-logging/loki-explore-query-systemd-rke2-server.png)
+
```

Comment:
If `Splunk` is tested, please add a chapter about how to and the output, thanks.